mpz/nextprime.c

```diff
@@ -82,47 +122,54 @@ mpz_nextprime (mpz_ptr p, mpz_srcptr n)
   /* Compute residues modulo small odd primes */
   moduli = TMP_SALLOC_TYPE (prime_limit * sizeof moduli[0], unsigned short);
 
-  for (;;)
-    {
-      /* FIXME: Compute lazily? */
```

Comment:
I tried this too and it's not useful, because to exit the lower loop you have to check each of these is non-zero, so there all need to be calculated, and reordering doesn't seem to help (as there is overhead in tracking if they are calculated)